src/framework/components/element/image-element.js

```diff
@@ -328,75 +328,92 @@ Object.assign(pc, function () {
-                        }
-                    } else {
-                        if (this.sprite) {
-                            if (this.sprite.renderMode === pc.SPRITE_RENDERMODE_SLICED) {
-                                this._material = this._system.defaultScreenSpaceImage9SlicedMaterial;
-                            } else if (this.sprite.renderMode === pc.SPRITE_RENDERMODE_TILED) {
-                                this._material = this._system.defaultScreenSpaceImage9TiledMaterial;
-                            } else {
-                                this._material = this._system.defaultScreenSpaceImageMaterial;
-                            }
-                        } else {
-                            this._material = this._system.defaultScreenSpaceImageMaterial;
-                        }
-                    }
-
-                }
+            var mask = !!this._mask;
+            var nineSliced = !!(this.sprite && this.sprite.renderMode === pc.SPRITE_RENDERMODE_SLICED);
+            var nineTiled = !!(this.sprite && this.sprite.renderMode === pc.SPRITE_RENDERMODE_TILED);
 
-                if (this._renderable) this._renderable.setCull(false);
-
-            } else {
-                if (!this._hasUserMaterial()) {
-                    if (this._mask) {
-                        if (this.sprite) {
-                            if (this.sprite.renderMode === pc.SPRITE_RENDERMODE_SLICED) {
-                                this._material = this._system.defaultImage9SlicedMaskMaterial;
-                            } else if (this.sprite.renderMode === pc.SPRITE_RENDERMODE_TILED) {
-                                this._material = this._system.defaultImage9TiledMaskMaterial;
-                            } else {
-                                this._material = this._system.defaultImageMaskMaterial;
-                            }
-                        } else {
-                            this._material = this._system.defaultImageMaskMaterial;
-                        }
-                    } else {
-                        if (this.sprite) {
-                            if (this.sprite.renderMode === pc.SPRITE_RENDERMODE_SLICED) {
-                                this._material = this._system.defaultImage9SlicedMaterial;
-                            } else if (this.sprite.renderMode === pc.SPRITE_RENDERMODE_TILED) {
-                                this._material = this._system.defaultImage9TiledMaterial;
-                            } else {
-                                this._material = this._system.defaultImageMaterial;
-                            }
-                        } else {
-                            this._material = this._system.defaultImageMaterial;
-                        }
-                    }
-                }
-
-                if (this._renderable) this._renderable.setCull(true);
+            if (!this._hasUserMaterial()) {
```

Comment:
Does this `hasUserMaterial` function work right now that materials are created later?

Reply:
I believe so. Materials are added to `defaultImageMaterials` in the system when they are created so if the default material exists hasUserMaterial() will correctly test against it.